gpu_ocean/SWESimulators/IPythonMagic.py

```diff
@@ -76,30 +88,69 @@ def exitfunc():
             self.logger.debug("==================================================================")
         atexit.register(exitfunc)
         
+        
+        
+        
+        
+        
+    logger_initialized = False
```

Comment:
This makes sure that if you use logging, but run the initialization magic twice, it only initializes once. Very handy